### PR TITLE
Initial import_playbook support

### LINKED ISSUE #109: Support import_playbook
Using `import_playbook:` causes `There was an error during execution: 'hosts'`. I suspect it's looking for a `hosts:` on `- import_playbook:`. I suggest either printing an error that it isn't supported or adding support for it. I'll work around it by turning the playbook into a role or tasks file.

build_test.yml:
```yaml
- import_playbook: test.yml
  vars:
    ansible_bender:
      base_image: python:3.5-stretch
      target_image:
        name: test_img
```

test.yml:
```yaml
- hosts: all
  tasks:
    - debug:
        msg: hi
```

Running ansible-bender:
```
$ ansible-bender -v build test_build.yml
14:35:02.769 utils.py          INFO   running command: "['ansible-playbook', '--version']"
14:35:03.334 utils.py          INFO   running command: "['ansible-playbook', '-c', 'local', '-i', '/tmp/ab96owsyrm/i', '-e', 'ansible_python_interpreter=/.../.virtualenvs/ansible-bender/bin/python', '.test_build-20190411143502768844-afiunkadkq.yaml']"
14:35:04.772 utils.py          INFO   running command: "['podman', 'version']"
14:35:04.817 utils.py          INFO   running command: "['buildah', 'version']"
14:35:04.852 utils.py          INFO   running command: "['podman', 'image', 'exists', 'python:3.5-stretch']"
14:35:04.986 utils.py          INFO   running command: "['podman', 'run', '--rm', 'python:3.5-stretch', 'true']"
14:35:08.050 utils.py          INFO   running command: "['buildah', 'inspect', '-t', 'image', 'python:3.5-stretch']"
14:35:08.211 utils.py          INFO   running command: "['podman', 'run', '--rm', 'python:3.5-stretch', 'ls', '/usr/bin/python3']"
14:35:11.403 buildah_builder.py INFO   using python interpreter /usr/bin/python3
14:35:11.403 utils.py          INFO   running command: "['buildah', 'from', '--name', 'test_img-20190411-143504761617-cont', '2bf898387bbd9f15e7f50eb4c1c491cd2dff636fd204f40a7f05e9aabd97375e']"
14:35:13.564 core.py           INFO   creating inventory file /tmp/abw5caf2xq/inventory
14:35:13.566 utils.py          INFO   running command: "['buildah', 'rm', 'test_img-20190411-143504761617-cont']"
There was an error during execution: 'hosts'
```

Environment:
```
$ pip freeze
ansible-bender==0.5.2.dev1+g29c8d01
attrs==19.1.0
jsonschema==3.0.1
pyrsistent==0.14.11
PyYAML==5.1
six==1.12.0
tabulate==0.8.3
$ python --version
Python 3.7.2
```

`import_tasks` does work. test_build2.yml:
```yaml
- hosts: all
  vars:
    ansible_bender:
      base_image: python:3.5-stretch
      target_image:
        name: test_img
  tasks:
    - import_tasks: test.yml
```

test2.yml:
```yaml
- debug:
    msg: hi
```

Thanks

### PR DESCRIPTION
**Pull request by themkat**
## Basics
Ignores adjusting hosts for import_playbooks statements, as those should not have a host entry. Instead we have to read the imported playbook and correct the possible hosts entries there (this goes on recursively by need, so can support a very big import_playbook-depth). The hosts-corrected imported playbook path/name will need to be used in place of the old one in the playbook that imports it. I decided on a uuid based filename for the temp files, because of the low likelihood of "name-crashes". Currently the temp-file was named p.yaml, and if we have several ones that would off course not work. This PR implements those fixes.

tl;dr: fixes #109 

## About the testing
I initially planned to do everything in tests_core.py, but after some thinking decided to create e2e'ish test in tests_api.py instead. This is because the tests in core would have needed either a lot of setup (duplication of api.py-code maybe), or mocking (in which I think it wouldn't be as useful). Hopefully this is not an issue, as I think the tests I made still verifies the functionality in an ok way. 

The tests tests the example from #109, as well as a recursive one (import_playbook inside an imported playbook). The last one might be overkills. I have also tested manually, and it seems to work as expected in the tests I've done so far 🙂

### CODE REVIEW

File: ansible_bender/core.py [hunk 1]
```diff
@@ -173,6 +174,29 @@ def _get_path_our_site(self):
         # hence, let's add the site ab is installed in to sys.path
         return os.path.dirname(os.path.dirname(ansible_bender.__file__))
 
+    def _correct_host_entries(self, playbook_path, tmpDir):
+        """ Correct the host entries in the playbook and all imported playbooks """
+        tmp_pb_path = os.path.join(tmpDir, str(uuid.uuid4()) + ".yaml")
```
Comment (TomasTomecek) [inline ansible_bender/core.py:179]: love using the uuid here! could we please prefix the uuid string with e.g. `ab_` so that people know the file is coming from ansible-bender?

File: ansible_bender/core.py [hunk 2]
```diff
@@ -173,6 +174,29 @@ def _get_path_our_site(self):
...
+
+        with open(playbook_path, "r") as fd_r:
+            pb_dict = yaml.safe_load(fd_r)
+            
+        for idx, doc in enumerate(pb_dict):
+            imported_playbook = doc.get("import_playbook")
+            if imported_playbook:
+                import_base_path = os.path.dirname(playbook_path)
+                imported_playbook_path = os.path.join(import_base_path, imported_playbook)
+                doc["import_playbook"] = self._correct_host_entries(imported_playbook_path, tmpDir)
```
Comment (TomasTomecek) [inline ansible_bender/core.py:189]: would some logger.debug make sense here as well? especially when it's called recursively

**Review by TomasTomecek — COMMENTED** [review]
wow, very nice work!

just a few nits and we're good to merge

File: ansible_bender/core.py (hunk 2 above)
Comment (themkat) [reply ansible_bender/core.py]: Made a simple one in [2abc606](https://github.com/ansible-community/ansible-bender/pull/268/commits/2abc6065f4735e122fda3513b9cfe32b7ade51f3) 🙂  Does it look okay?

Comment (TomasTomecek) [reply ansible_bender/core.py]: yup, looks fine!

**Comment by TomasTomecek** [comment]
merging, thank you so much for taking time to improve bender!

I'll try to fix some of the test runs in a separate PR so we can get green CI
